kanin/src/extract.rs

```diff
@@ -30,36 +30,18 @@ pub trait Extract<S>: Sized {
     async fn extract(req: &mut Request<S>) -> Result<Self, Self::Error>;
 }
 
-/// Note that when you extract the [`Delivery`], the handler itself must acknowledge the request.
-/// Kanin *will not* acknowledge the request for you in this case.
-#[async_trait]
-impl<S> Extract<S> for Delivery
-where
-    S: Send + Sync,
-{
-    type Error = HandlerError;
-
-    async fn extract(req: &mut Request<S>) -> Result<Self, Self::Error> {
-        req.delivery
-            .take()
-            .ok_or(HandlerError::DELIVERY_ALREADY_EXTRACTED)
-    }
-}
-
 /// Note that when you extract the [`Acker`], the handler itself must acknowledge the request.
 /// kanin *will not* acknowledge the request for you in this case.
+// TODO: This implementation is quite hacky and should probably be removed.
```

Comment:
Now that we have the option to set timeout on queues?

Reply:
Yes